### ISSUE TITLE
[MaterialDatePicker] Added Support for a Spinner Mode

### ISSUE THREAD

**Issue by froriz5** [material-components/material-components-android#1264]
### Context:
One of the modes supported from the older AppCompat Date Pickers was a Spinner mode.

This mode allowed users to use a set of dialers/spinners to select individual year/month/day directly.

<img width="896" alt="DatePickerSpinner" src="https://user-images.githubusercontent.com/1795695/80824707-9a6db580-8ba4-11ea-8f13-bfc4c4f60203.png">

Source: https://developer.android.com/guide/topics/ui/controls/pickers

It would be great to have `MaterialDatePicker` support such a mode, to align with use cases where it because difficult for a user to input an exact date, without having to scroll back in the Calendar View or Input a Text Entry Field.

### Example Use Case:
Is a user wants to select a particular month, from a given range of months:
<img width="148" alt="Screen Shot 2020-05-01 at 11 48 03 AM" src="https://user-images.githubusercontent.com/1795695/80825013-2e3f8180-8ba5-11ea-8a00-c39bb0dd7c25.png">

In this example, if the user wants to select `March 2018`, having a spinner to do so, would lead to a faster experience than the Calendar View.

**Describe alternatives you've considered**
The alternative is to use the old `DatePickerDialog` which has support for the Spinner Mode.

**Additional context**
Source: https://developer.android.com/guide/topics/ui/controls/pickers

**Comment by wcshi** [comment]
This is not part of the Material Design spec so we will unlikely be adding support. Could you please explain why it's better than using the input field?

**Comment by froriz5** [comment]
I think that the Input Field can accomplish this same goal, but I have a few issues with how it's implemented.

### From a UX point of view:
I find the experience of trying to modify a Text Input Field to be a more clunky experience, such as modifying individual params to change the date.

If I set the date to an initial selection of `4/1/2020`, but then the user want to modify the `day` or `month`, they have to move the IME cursor to the appropriate substring area.

![editingInputText](https://user-images.githubusercontent.com/1795695/80827122-03efc300-8ba9-11ea-8966-1f741972c520.png)

### From a Technical point of view:
It seems that the Text Input Field still allows the user to select an invalid value and click the `OK` Positive Button. This maybe a bug with the current picker.

![inputFieldValidation](https://user-images.githubusercontent.com/1795695/80827137-0b16d100-8ba9-11ea-8418-c5edd4da687d.png)

I know for the Calendar View, the user is not allowed to select a value that fails the `CalendarConstraints.DateValidator` set to the picker.

This doesn't seem to be the case for the Text Input Field.

**Comment by mcastagnolo** [comment]
> This is not part of the Material Design spec so we will unlikely be adding support. Could you please explain why it's better than using the input field?

The input field comes with no indication as to what you're supposed to type until you enter the first character, upon which you're treated to a glaring error field that finally tells you what you need to do. Then you type out the rest of your date, including forward slashes, fixing any mistypes, until you've finally satisfied the input field's requirements and it no longer shows you an error.

That experience clearly pales in comparison to just showing a spinner dialog and flicking a few inputs. 

There's a reason why nearly every Google app has a custom implementation of a date spinner: it's a clean, discoverable, and user-friendly experience.

**Comment by gabrielemariotti** [comment]
@mcastagnolo 

>The input field comes with no indication as to what you're supposed to type until you enter the first character

With the PR #1528 and the 6050bd4 commit the inputfield has a placeholder.
![image](https://user-images.githubusercontent.com/2583078/94335794-35e14600-ffde-11ea-84fe-42e29f92875c.png)

>Then you type out the rest of your date, including forward slashes, fixing any mistypes, until you've finally satisfied the input field's requirements and it no longer shows you an error.

With the 5541f02 commit there is a delay in the validation which should solve this issue.

Note that these commits are currently available only in the snapshot release.

**Comment by G00fY2** [comment]
The old system spinner mode is actually still working pretty fine (also together with MDC): https://github.com/G00fY2/SpinnerDatePicker

Just as hint for people may waiting for a new implementation.

**Comment by mobiRic** [comment]
Came here to find that my issue already exists, but is almost 2 years old.

It was asked _why_ one would prefer the old style:
- I came here because my real world users are consistently entering incorrect dates of birth into a real world medical app.
(the app was originally built with `MaterialDatePicker`)

This is direct User Experience that I'm seeing; and these users are having a very poor experience, so we are tossing this control from our app. Being a major fan of Material code, I'd much rather use this library than something else, but that is not an option.

tl;dr - the reason is that users don't know how to use this control

**Comment by ruzicka1911** [comment]
I came here because my real world users are working in the warehouses, where they need to enter expiration dates. The effectivity is worse when they use the new MaterialDatePicker versus the old one with spinners. Could you also add support for the spinners to new MaterialDatePicker. Thank you.
